### PR TITLE
ci: setup release and publish

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,3 +44,47 @@ jobs:
 
       - name: test
         run: cargo test --locked --all-targets
+
+      - name: Pre-release (linux)
+        if: |
+          startsWith(matrix.config.os, 'ubuntu')
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:50]:
```suggestion
          startsWith(matrix.os, 'ubuntu')
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,3 +44,47 @@ jobs:
...
+        run: |
+          cd target/release
+          zip -r deno_lint-x86_64-unknown-linux-gnu.zip deno_lint
+
+      - name: Pre-release (mac)
+        if: |
+          startsWith(matrix.config.os, 'macOS')
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:57]:
```suggestion
          startsWith(matrix.os, 'macOS')
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -44,3 +44,47 @@ jobs:
...
+        run: |
+          cd target/release
+          zip -r deno_lint-x86_64-apple-darwin.zip deno_lint
+
+      - name: Pre-release (windows)
+        if: |
+          startsWith(matrix.config.os, 'windows')
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:64]:
```suggestion
          startsWith(matrix.os, 'windows')
```

**Comment by bartlomieju** [comment]
Oooops, thanks @lucacasonato

**Pull request merged**